src/device.cc

```diff
@@ -238,14 +264,53 @@ std::shared_ptr<SubHidrawConnection> SubHidrawConnection::create(const DeviceSca
     connection->m_details.deviceFlags |= DeviceFlag::Vibrate;
   }
 
-  // Create socket notifier
-  connection->m_notifier = std::make_unique<QSocketNotifier>(devfd, QSocketNotifier::Read);
-  QSocketNotifier* const notifier = connection->m_notifier.get();
+  // Create read and write socket notifiers
+  connection->m_readNotifier = std::make_unique<QSocketNotifier>(devfd, QSocketNotifier::Read);
+  QSocketNotifier* const readNotifier = connection->m_readNotifier.get();
   // Auto clean up and close descriptor on destruction of notifier
-  connect(notifier, &QSocketNotifier::destroyed, [notifier]() {
-    ::close(static_cast<int>(notifier->socket()));
+  connect(readNotifier, &QSocketNotifier::destroyed, [readNotifier]() {
+    ::close(static_cast<int>(readNotifier->socket()));
+  });
+
+  connection->m_writeNotifier = std::make_unique<QSocketNotifier>(devfd, QSocketNotifier::Write);
+  QSocketNotifier* const writeNotifier = connection->m_writeNotifier.get();
+  // Auto clean up and close descriptor on destruction of notifier
+  connect(writeNotifier, &QSocketNotifier::destroyed, [writeNotifier]() {
+    ::close(static_cast<int>(writeNotifier->socket()));
   });
 
   connection->m_details.phys = sd.phys;
+  connection->disableWrite(); // disable write notifier
   return connection;
 }
+
+// -------------------------------------------------------------------------------------------------
+ssize_t SubDeviceConnection::sendData(QByteArray hidppMsg)
+{
+  ssize_t res = -1;
+  bool isValidMsg = (hidppMsg.length() == 7 && hidppMsg.at(0) == 0x10);             // HID++ short message
+  isValidMsg = isValidMsg || (hidppMsg.length() == 20 && hidppMsg.at(0) == 0x11);   // HID++ long message
+
+  if (type() == ConnectionType::Hidraw && mode() == ConnectionMode::ReadWrite
+          && m_writeNotifier && isValidMsg)
+  {
+    enableWrite();
+    const auto notifier = socketWriteNotifier();
+    res = ::write(notifier->socket(), hidppMsg.data(), hidppMsg.length());
+    logInfo(hid) << "Write " << hidppMsg.toHex();
```

Comment:
Should be `logDebug` instead of `logInfo` also more detailed: e.g.:
```suggestion
    logDebug(hid) << "Write " << hidppMsg.toHex() << "from" << connection.path(); 
```

Reply:
Hi! the change has been implemented.